testing/go/enginetest/doltgres_harness.go

```diff
@@ -0,0 +1,516 @@
+			runQuery, sanitized := sanitizeQuery(s)
+			if !runQuery {
+				t.Log("Skipping setup query: ", s)
+				continue
+			} else {
+				t.Log("Running setup query: ", s)
+			}
+			_, rowIter, err := queryEngine.Query(ctx, sanitized)
+			if err != nil {
+				return nil, err
+			}
+			err = drainIter(ctx, rowIter)
+			if err != nil {
+				return nil, err
+			}
+		}
+	}
+
+	return queryEngine, nil
+}
+
+var skippedSetupWords = []string{
+	"auto_increment",
+	"create index",
+	"bigtable",           // "ERROR: blob/text column 't' used in key specification without a key length"
+	"typestable",         // lots of work to do
+	"datetime_table",     // invalid timestamp format
+	"specialtable",       // invalid quoting
+	"people",             // ERROR: blob/text column 'first_name' used in key specification without a key length
+	"reservedWordsTable", // ERROR: blob/text column 'Timestamp' used in key specification without a key length
+	"foo.othertable",     // ERROR: database schema not found: foo (errno 1105)
+	"bus_routes",         // ERROR: blob/text column 'origin' used in key specification without a key length
+	"parts",              // ERROR: blob/text column 'part' used in key specification without a key length
+	"xy_hasnull_idx",     // needs an index during creation
+	"xy ",                // needs an index during creation
+	"rs ",                // needs an index during creation
+	"analyze table",      // unsupported syntax
+}
+
+var commentClause = regexp.MustCompile(`(?i)comment '.*?'`)
+var createIndexStatement = regexp.MustCompile(`(?i)create.*?index`)
+var alterTableStatement = regexp.MustCompile(`(?i)alter table`)
+var createTableStatement = regexp.MustCompile(`(?i)create table`)
+var floatKeyword = regexp.MustCompile(`(?i)\bfloat\b`)
+var doubleKeyword = regexp.MustCompile(`(?i)\bdouble\b`)
+var datetimeKeyword = regexp.MustCompile(`(?i)\bdatetime\b`)
+var mediumIntKeyword = regexp.MustCompile(`(?i)\bmediumint\b`)
+var tinyIntKeyword = regexp.MustCompile(`(?i)\btinyint\b`)
+var backtick = "`"
+
+// sanitizeQuery strips the query string given of any unsupported constructs without attempting to actually convert
+// to Postgres syntax.
+func sanitizeQuery(s string) (bool, string) {
+	for _, word := range skippedSetupWords {
+		if strings.Contains(s, word) {
+			return false, ""
+		}
+	}
+
+	if createIndexStatement.MatchString(s) {
+		return false, ""
+	}
+	if alterTableStatement.MatchString(s) {
+		return false, ""
+	}
+
+	if createTableStatement.MatchString(s) {
+		s = replaceTypes(s)
+	}
+
+	s = commentClause.ReplaceAllString(s, "")
+	s = strings.ReplaceAll(s, backtick, `"`)
+
+	return true, s
+}
+
+func replaceTypes(s string) string {
+	s = floatKeyword.ReplaceAllString(s, "real")
+	s = doubleKeyword.ReplaceAllString(s, "double precision")
+	s = datetimeKeyword.ReplaceAllString(s, "timestamp")
+	s = mediumIntKeyword.ReplaceAllString(s, "integer")
+	s = tinyIntKeyword.ReplaceAllString(s, "smallint") // TODO: bool?
```

Comment:
Definitely seems like it should be `smallint`. `tinyint` is sometimes used to "represent" booleans, but since it's not an actual requirement in GMS, we shouldn't treat it as such.